ddtrace/tracer/example_test.go

```diff
@@ -3,37 +3,47 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016 Datadog, Inc.
 
-package tracer
+package tracer_test
 
 import (
-	"io/ioutil"
-	"log"
+	"context"
+	"fmt"
+	"time"
 
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 )
 
 // A basic example demonstrating how to start the tracer, as well as how
 // to create a root span and a child span that is a descendant of it.
 func Example() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
 	// Start the tracer and defer the Stop method.
-	Start(WithAgentAddr("host:port"))
-	defer Stop()
+	tracer.Start()
+	defer tracer.Stop()
 
 	// Start a root span.
-	span := StartSpan("get.data")
+	span, ctx := tracer.StartSpanFromContext(ctx, "parent")
 	defer span.Finish()
 
-	// Create a child of it, computing the time needed to read a file.
-	child := StartSpan("read.file", ChildOf(span.Context()))
-	child.SetTag(ext.ResourceName, "test.json")
+	// Run some code.
+	doSomething(ctx)
+}
+
+func doSomething(ctx context.Context) {
+	// Create a child, using the context of the parent span.
+	span, ctx := tracer.StartSpanFromContext(ctx, "do.something", tracer.Tag(ext.ResourceName, "alarm"))
+	defer func() {
+		span.Finish(tracer.WithError(ctx.Err()))
+	}()
```

Comment:
Wouldn't it be better to use an example with a common use case, which is usually an `error` variable (and potentially a returned value of this function) ? Such as:
```suggestion
	defer func(err error) {
		span.Finish(tracer.WithError(err))
	}(err)
```
This is a much more common case than using `ctx.Err()`. It's also a bit odd that the function triggers an error but doesn't let the caller know, yet it is important enough for tracing.

Reply:
Sure, returning an error and using that instead sounds good, especially if that's the most common use case.

@ajgajg1134 discussed the problems with passing in the error message as a param to the function in https://github.com/DataDog/dd-trace-go/pull/1389#discussion_r928933907
I confirmed that the way it's setting the error in the newest commit works as expected by running it locally 👍